app/public/js/common/SCapiService.js

```diff
@@ -186,6 +186,86 @@ app.service('SCapiService', function($http, $window, $q, $log, $state, $statePar
     };
 
     /**
+     * Responsible to get track details
+     * @return {[object]} data response
+     */
+    this.getTrackDetails = function(songId) {
+        var url = 'https://api.soundcloud.com/tracks/' + songId + '.json?&oauth_token=' + $window.scAccessToken,
+            that = this;
+        return $http.get(url)
+            .then(function(response) {
+                if (typeof response.data === 'object') {
+                    return response.data;
+                } else {
+                    // invalid response
+                    return $q.reject(response.data);
+                }
+
+            }, function(response) {
+                // something went wrong
+                return $q.reject(response.data);
+            });
+    };
+
+
+    /**
+     * Responsible to get comments for track
+     * @return {[object]} data response
+     */
+    this.getTrackComments = function(songId) {
+        var url = 'https://api.soundcloud.com/tracks/' + songId + '/comments.json?linked_partitioning=1&limit=30&oauth_token=' + $window.scAccessToken,
+            that = this;
+        return $http.get(url)
+            .then(function(response) {
+                if (typeof response.data === 'object') {
+                    if ( response.data.next_href !== null || response.data.next_href !== undefined ) {
+                        that.next_page = response.data.next_href;
+                    }
+                    return response.data;
+                } else {
+                    // invalid response
+                    return $q.reject(response.data);
+                }
+
+            }, function(response) {
+                // something went wrong
+                return $q.reject(response.data);
+            });
+    };
+
+    /**
+     * Responsible to get user favorite tracks ids
+     * @return {[object]} data response
+     */
+    this.getFavoritesIds = function(next_href) {
```

Comment:
Exactly nothing will happen, track won't be marked as liked when it's rather important information about track (on stream view - that's where it's always null).
As well in future cached favs list can be used to show on player bar if track was already liked.
Few lines of code isn't much.